rhalphalib/model.py

```diff
@@ -363,23 +363,35 @@ def autoMCStats(self, epsilon=0, threshold=0, include_signal=0, channel_name=Non
         first_sample = self._samples[list(self._samples.keys())[0]]
 
         for i in range(first_sample.observable.nbins):
-            ntot, etot2 = 0, 0
+            ntot_bb, etot2_bb = 0, 0  # for the decision to use bblite or not
+            ntot, etot2 = 0, 0  # for the bblite uncertainty
 
             # check if neff = ntot^2 / etot2 > threshold
             for sample in self._samples.values():
+                ntot += sample._nominal[i]
+                etot2 += sample._sumw2[i]
+
                 if not include_signal and sample._sampletype == Sample.SIGNAL:
                     continue
 
-                ntot += sample._nominal[i]
-                etot2 += sample._sumw2[i]
+                ntot_bb += sample._nominal[i]
+                etot2_bb += sample._sumw2[i]
 
             if etot2 <= 0.:
                 continue
+            elif etot2_bb <= 0:
+                # this means there is signal but no background, so create stats unc. for signal only
+                for sample in self._samples.values():
+                    if sample._sampletype == Sample.SIGNAL:
+                        sample_name = None if channel_name is None else channel_name + "_" + sample._name[sample._name.find('_') + 1:]
+                        sample.autoMCStats(epsilon=epsilon, sample_name=sample_name, bini=i)
+
+                continue
 
-            neff = ntot ** 2 / etot2
-            if neff <= threshold:
+            neff_bb = ntot_bb ** 2 / etot2_bb
+            if neff_bb <= threshold:
                 for sample in self._samples.values():
-                    sample_name = None if channel_name is None else channel_name + "_" + sample._name
+                    sample_name = None if channel_name is None else channel_name + "_" + sample._name[sample._name.find('_') + 1:]
```

Comment:
Doesn't https://github.com/nsmith-/rhalphalib/blob/f51e110b80e6097d58d820e69e9a745062802923/rhalphalib/model.py#L159 guarantee that the name is already fully qualified?

Reply:
Sorry, I don't understand. The `channel_name` here option is only in case the user wants a specific channel name for their MC stats parameters - otherwise it will default to the `{channel_name}_{sample_name}`. For context, this came up when I wanted to tie together the MC stats parameters in two versions of my pass region (blinded MC and unblinded MC), for which I had to use the same channel name for both sets of MC stats parameters. Does that make sense?

Reply:
Ok, I see.. because you have blinded and unblinded in the same workspace (which is a bit unusual) you need the common parameters tied together. Would an alternative be to have them be separate but just freeze the ones in the masked channel?

Reply:
I think not exactly, because we’re doing the fit in the “blinded” channels and then plotting the “unblinded” with the post-fit nuisance parameters, using channel masking. So we would need the parameters to be the same to retain the post-fit values. I think we’re abusing the blinded and unblinded terminology here: in both channels the data in the Higgs mass window is blinded; the difference is that for the fit we’re setting the MC to 0 in those bins as well, but we want to show it in the plots. So “blinded” and “unblinded” here only refers to the MC in the mass window. 

This is the strategy they did for boosted HH4b, which I guess is unusual but the only alternative I can think of is to do the fit with MC 0 and then copy over the post-fit nuisance values into the “unblinded” workspace?

Reply:
Yeah, I would have done it that way (render separate blinded and unblinded models, and copy the fit result over) but I can imagine then you don't get to use the combine built-in postfit template generation. So ok, then I see why this `channel_name` argument is here but it wasn't obvious to me until you explained the use case.